### ISSUE TITLE
Kotest Gradle plugin is missing in snapshot publications

### ISSUE THREAD

**Issue by OliverO2** [kotest/kotest#4012]
The `io.kotest.multiplatform` Gradle plugin (`io.kotest:kotest-framework-multiplatform-plugin-gradle`) is not published to the [snapshot repo](https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/). This complicates configuring Kotest dependencies (separate version for plugin and libraries required) and blocks using snapshots which require an updated plugin.

**Comment by OliverO2** [comment]
Maybe adding the last line of this to `.github/workflows/master.yml` would suffice?
```
jobs:
   linux:
      strategy:
         matrix:
            target:
               - :kotest-framework:kotest-framework-multiplatform-plugin-gradle:publish
```

**Comment by OliverO2** [comment]
@aSemy Seems like the Gradle plugin is still not published: https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/kotest-framework-multiplatform-plugin-gradle

**Comment by aSemy** [comment]
It looks like publishing in general doesn't work at the moment.

There are no new Kotest snapshots published in July

<img width="70%" alt="image" src="https://github.com/user-attachments/assets/efdc4102-7324-4c1f-a8d8-91b897bf5144">


https://github.com/kotest/kotest/actions/runs/9915068638/job/27396015667

```
Execution failed for task ':kotest-bom:publishKotestBomPublicationToDeployRepository'.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
> Failed to publish publication 'KotestBom' to repository 'deploy'
   > Could not PUT 'https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/kotest-bom/5.10.0.1526-SNAPSHOT/maven-metadata.xml'. Received status code 401 from server: Content access is protected by token
```

I think Sonatype broke publishing https://community.sonatype.com/t/401-content-access-is-protected-by-token-authentication-failure-while-performing-maven-release/12741

Anyway, it's good to re-open this issue so we can fix it.

**Comment by OliverO2** [comment]
If I look at [kotest-framework-engine](https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/kotest-framework-engine/), the last publication was [5.10.0.1487-SNAPSHOT/] on Wed Jun 12 07:27:57 UTC 2024.

So seemingly I confused this with July 12 (which should have included the fix in #4136). Should we close this one and open another issue on publishing in general?

**Comment by OliverO2** [comment]
And it seems like the last successful master CI run was on the 4th of June, and this is required for publishing a snapshot, so we should probably get the remaining flakiness fixed first.

@aSemy I've added a change to #4149 which addresses your concerns so let's proceed with that right away?

**Comment by aSemy** [comment]
Yes, I'll look at your changes in #4149 now, sorry for the delay.

**Comment by OliverO2** [comment]
No problem, just pinging because GH does not offer "re-request a review" when a review is still pending.

**Comment by aSemy** [comment]
Sorry, something came up! Will look sooner.

**Comment by OliverO2** [comment]
Closing in favor of #4178.

**Comment by aSemy** [comment]
Hmmm Sonatype doesn't like something about the plugin marker...

https://github.com/kotest/kotest/actions/runs/10017676112/job/27694238728#step:7:2337

```
> Task :kotest-framework:kotest-framework-multiplatform-plugin-gradle:publishKotestMultiplatformCompilerGradlePluginPluginMarkerMavenPublicationToDeployRepository FAILED


FAILURE: Build failed with an exception.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.


* What went wrong:
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
Execution failed for task ':kotest-framework:kotest-framework-multiplatform-plugin-gradle:publishKotestMultiplatformCompilerGradlePluginPluginMarkerMavenPublicationToDeployRepository'.

> Failed to publish publication 'KotestMultiplatformCompilerGradlePluginPluginMarkerMaven' to repository 'deploy'
   > Could not PUT 'https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/multiplatform/io.kotest.multiplatform.gradle.plugin/5.10.0.1533/io.kotest.multiplatform.gradle.plugin-5.10.0.1533.pom'. Received status code 400 from server: Bad Request

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org./

BUILD FAILED in 16m 24s
For more on this, please refer to https://docs.gradle.org/8.7/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
1770 actionable tasks: 1505 executed, 265 from cache

Publishing build scan...
https://gradle.com/s/mv2xfqfi6bcia
```

[build scan](https://scans.gradle.com/s/mv2xfqfi6bcia)


Possibly it's missing the Javadoc JAR.

**Comment by aSemy** [comment]
Publishing still doesn't work.

https://github.com/kotest/kotest/actions/runs/10020863926/job/27699301228#step:8:3222

```
> Task :kotest-framework:kotest-framework-multiplatform-plugin-gradle:publishKotestMultiplatformCompilerGradlePluginPluginMarkerMavenPublicationToDeployRepository FAILED


FAILURE: Build failed with an exception.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.


* What went wrong:
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
Execution failed for task ':kotest-framework:kotest-framework-multiplatform-plugin-gradle:publishKotestMultiplatformCompilerGradlePluginPluginMarkerMavenPublicationToDeployRepository'.

> Failed to publish publication 'KotestMultiplatformCompilerGradlePluginPluginMarkerMaven' to repository 'deploy'
For more on this, please refer to https://docs.gradle.org/8.7/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
   > Could not PUT 'https://s01.oss.sonatype.org/content/repositories/snapshots/io/kotest/multiplatform/io.kotest.multiplatform.gradle.plugin/5.10.0.1545/io.kotest.multiplatform.gradle.plugin-5.10.0.1545.pom'. Received status code 400 from server: Bad Request
2031 actionable tasks: 1878 executed, 153 from cache

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org./

BUILD FAILED in 24m 55s

Publishing build scan...
https://gradle.com/s/gdcgonb3kumxw
```

I think this is because Sonatype wants the group ID to be `io.kotest`, but for the plugin-marker it's the plugin ID, which is `io.kotest.multiplatform`, and Sonatype doesn't automatically see them as the same.

I'll send an email to Sonatype.

**Comment by aSemy** [comment]
Ohhhh I think the issue is the Kotest GP is missing a `-SNAPSHOT` suffix. I'll make a PR...

**Comment by OliverO2** [comment]
Fixed by #4194. Yay!
